octobot_trading/personal_data/portfolios/portfolio_manager.py

```diff
@@ -227,15 +254,30 @@ def _refresh_simulated_trader_portfolio_from_order(self, order):
                                   f"for order {order.to_dict()}")
         return False
 
-    def _load_portfolio(self):
+    def _load_portfolio(self, reset_from_config):
         """
         Load simulated portfolio from config if required
         """
         if self.trader.is_enabled:
             if self.trader.simulate:
-                self._set_starting_simulated_portfolio()
+                if reset_from_config \
+                        or self.historical_portfolio_value_manager is None \
+                        or not self.historical_portfolio_value_manager.has_previous_session_portfolio():
+                    self._set_starting_simulated_portfolio()
+                else:
+                    self._load_simulated_portfolio_from_history()
             self.logger.info(f"{constants.CURRENT_PORTFOLIO_STRING} {self.portfolio.portfolio}")
 
+    def _load_simulated_portfolio_from_history(self):
+        #  todo also load available amounts when loading simulated orders
```

Comment:
Is it done now with `_apply_locked_assets`?

Reply:
we will be able to do it when simulated orders will be saved in db and reloaded at startup